src/constructs/core/stack.ts

```diff
@@ -13,8 +13,16 @@ import type { GuMappingValue, GuStageMappingValue } from "./mappings";
 import { GuStageParameter } from "./parameters";
 import type { GuParameter } from "./parameters";
 
-export interface GuStackProps extends StackProps, Partial<GuMigratingStack> {
+export interface GuStackProps extends Omit<StackProps, "stackName">, Partial<GuMigratingStack> {
+  /**
+   * The Guardian stack being used (as defined in your riff-raff.yaml).
+   * This will be applied as a tag to all of your resources.
+   */
   stack: string;
```

Comment:
It was tempting to rename this too, but:

a) This would mean updating every single stack using `@guardian/cdk`, which is a lot of effort
b) The tag that gets applied is still called `Stack` and in the riff-raff.yaml the property is called `stacks`, so we can't fully escape from this ambiguity anyway